src/config.rs

```diff
@@ -139,7 +128,29 @@ impl Config {
         Ok(self)
     }
 
+    /// Pushes new [`Source`] to Config unless it is frozen
+    #[deprecated(since = "0.12.0", note = "please use 'ConfigBuilder' instead")]
+    pub(crate) fn push<T>(&mut self, source: T) -> Result<()>
+    where
+        T: 'static,
+        T: Source + Send + Sync 
+    {
+        match self.kind {
+            ConfigKind::Mutable {
+                ref mut sources, ..
+            } => {
+                sources.push(Box::new(source));
+                return Ok(())
+            }
+
+            ConfigKind::Frozen => {
+                return Err(ConfigError::Frozen);
+            }
+        }
+    }
```

Comment:
Please also remove this function when cleaning up my above comments.

Reply:
Done.